plugins/inputs/intel_pmu/README.md

```diff
@@ -0,0 +1,196 @@
+# Intel Performance Monitoring Unit Plugin
+
+This input plugin exposes Intel PMU (Performance Monitoring Unit) metrics available through [Linux Perf](https://perf.wiki.kernel.org/index.php/Main_Page) subsystem.
+
+PMU metrics gives insight into performance and health of IA processor's internal components,
+including core and uncore units. With the number of cores increasing and processor topology getting more complex
+the insight into those metrics is vital to assure the best CPU performance and utilization.
+
+Performance counters are CPU hardware registers that count hardware events such as instructions executed, cache-misses suffered, or branches mispredicted.
+They form a basis for profiling applications to trace dynamic control flow and identify hotspots.
+
+### Configuration:
+```toml
```

Comment:
Make sure to follow the config style guide (https://github.com/influxdata/telegraf/blob/master/docs/developers/SAMPLE_CONFIG.md#style) It looks like you have some config values here that aren't applicable to all users or are defaults that should be commented.

Sync with intel_pmu.go SampleConfig() if you make changes.

Reply:
In terms of defaults, I believe that all of them are commented out. If you mean "events" or "cores"/"sockets" fields, there is a quote in guidance: "If a parameter does not have a default **or must frequently be changed then have it uncommented**". So there are parameters that need to be frequently changed like a mentioned list of events or the list of cores/sockets. They do not have a specific default value, only we check if the parameters are absent in config, we consider it as a will to obtain all possible events/cores/sockets, but this is not a very common case.

Could you point out which exactly config values here aren't applicable to all users or don't meet the guidelines? Thanks!

Reply:
They change frequently and that's the most common use case for the plugin is to fill this with event names that need to be counted - so we'd keep that uncommented. But for sure we could explicitly state that fact in the comment above events field.

Added "Event names shall match names from event_definitions files" info.